docs/src/guides/architecture-patterns.md

```diff
@@ -0,0 +1,92 @@
+// for modern web frameworks to track changes in data, you will want to
+// return the result of the update function vs just the object you passed in.
+```
+
+### Working with Modern Web Frameworks
+
+All modern web frameworks (React, Angular, Vue, Ember etc) leverage some form of change tracking to determine when to re-render the UI. Typically this is done via object equality (aka `===`) or ES6 proxy objects, both of which are very efficient, but add complexity when working with Class instances. To keep things simple for all frameworks, Hub.js has enforced separation between the Manager classes and the Data. All Manager functions which make changes to the data, will return a new, updated instance of the data.
+
+### Composition via Interfaces and Functions
+
+Manager classes and the POJO's they work with are defined via interface composition. The Managers all implement the `IHubEntityManager` interface, which defines the basic CRUD+Search operations.
+
+Additionally, managers will also implement "trait manager interfaces" (i.e. `IWithLayoutManager`), which define behaviors which the manager will provide implementations for. This allows us to share both structure and behavior across many different entities, while still enabling per-entity customization where necessary.
+
+Similarly, the POJO's will conform to the entity specific interface (i.e. `IHubProject`) but those interfaces are composed with "trait interfaces" to ensure consistent properties are used across the types.
+
+For example `IHubProject` extends the `IWithSlug` iterface, and the `HubProjectManager` implements the `IWithSlugManager` interface, operating on the properties defined by `IWithSlug`.
+
+![Manager and Pojo Interfaces](/hub.js/img/interface-hiearchy.png)
+
+This pattern ensures consistent API's across entities, powered by stand alone functions, composed into the Manager Classes and data objects.
+
+## Hub Automation Class
+
+The final pattern is oriented towards developers looking to automate various Hub activities via scripting. In this pattern, all the core Managers are accessible from a central [`Hub` class](/hub.js/api/common/Hub).
+
+```js
+import { Hub } from "@esri/hub-common";
+
+// create Hub instance
+const myHub = await Hub.create({
+  authOptions: { username: "casey", password: "abc123" },
+});
+
+// create a project via the HubProjectManager, which is accessed via `.projects`
+let project = myHub.projects.create({
+  name: "Smith Street Curb Replacement",
+  summary: "Replace/upgrade damaged curbs along Smith St. Scheduled for 2024",
+  status: "active",
+  culture: "en-us",
+  slug: "smith-st-2024",
+  tags: ["2024 Plan"],
+});
+console.info(`Created project ${project.name} owned by ${project.owner}`);
+```
+
+While this api could be used in a web application, the tight coupling between the Hub class and all the Manager classes means that there would be significant overhead in any build output.
```

Comment:
What type of "overhead"? (Size, processing, parsing, cognitive,...

Reply:
added